Source/WebCore/html/ImageBitmap.cpp

```diff
@@ -544,7 +545,13 @@ void ImageBitmap::createCompletionHandler(ScriptExecutionContext& scriptExecutio
     }
 
     auto outputSize = outputSizeForSourceRectangle(sourceRectangle.returnValue(), options);
-    auto bitmapData = createImageBuffer(scriptExecutionContext, outputSize, bufferRenderingMode(scriptExecutionContext), imageForRender->colorSpace());
+    auto renderingMode = bufferRenderingMode(scriptExecutionContext);
+#if USE(SKIA)
+    // Make sure ImageBitmap is unaccelerated when created from unaccelerated canvas.
+    if (renderingMode == RenderingMode::Accelerated && canvas.renderingContext() && !canvas.renderingContext()->isGPUBased())
+        renderingMode = RenderingMode::Unaccelerated;
+#endif
+    auto bitmapData = createImageBuffer(scriptExecutionContext, outputSize, renderingMode, imageForRender->colorSpace());
```

Comment:
So, what matters here is not that the canvas is accelerated, but that the image we are going to draw is accelerated. Of course, the image is accelerated because the canvas is accelerated in this case, but I wonder if we can generalize this by checking if the image is accelerated instead, and do the same in other constructors.

Reply:
In some cases, the fact that the image is accelerated is fine. In the testing I did, I discovered that only the case of accelerated `ImageBuffer` being created from unaccelerated canvas is problematic and triggers "noop" path in skia.